src/GeolocationContext.tsx

```diff
@@ -118,21 +118,69 @@ export const GeolocationContextProvider: React.FC = ({children}) => {
   async function requestPermission() {
     if (!(await isLocationEnabled())) {
       Alert.alert(
-        'Du har blokkert posisjonsdeling',
-        'For å kunne bruke posisjonen din må du aktivere lokasjonstjenester på telefonen din.',
+        t(GeoLocationTexts.blockedLocation.title),
+        t(GeoLocationTexts.blockedLocation.message),
       );
-      dispatch({
-        type: 'PERMISSION_CHANGED',
-        status: state.status,
-        locationEnabled: false,
-      });
     } else {
       const status = await requestGeolocationPermission();
       dispatch({type: 'PERMISSION_CHANGED', status, locationEnabled: true});
       return status;
     }
   }
 
+  async function requestGeolocationPermission(): Promise<PermissionStatus> {
```

Comment:
I might consider moving permissions into src/utils as a hook and have it there to reuse in later views.

@marius-at-atb made a placeholder https://github.com/AtB-AS/mittatb-app/blob/3d115b3843b89ed48535707d67fc372345432917/src/utils/permissions.ts that might be inspire you.